OpenCL_C.txt

```diff
@@ -3178,10 +3178,13 @@ The following predefined macro names are available.
     source line (an integer constant).
 
 `+__OPENCL_VERSION__+` ::
-    Substitutes an integer reflecting the version number of the OpenCL
-    supported by the OpenCL device.
-    The version of OpenCL described in this document will have
-    `+__OPENCL_VERSION__+` substitute the integer 300.
+    For OpenCL devices with OpenCL version less than or equal to OpenCL 2.0,
+    substitutes an integer value reflecting the OpenCL version supported by the
+    device.
+    This predefined macro is <<unified-spec, deprecated by>> OpenCL 2.1.
+    For OpenCL devices with OpenCL version greater than OpenCL 2.0, it must be
+    defined but may substitute any implementation-defined integer value greater
+    than OpenCL 2.0. footnote:[{fn-OPENCL_VERSION}]
```

Comment:
We should say how the version is encoded, i.e `Mmp` with `M` the major version, `m` the minor version and `p` the patch version.

Being pedantic "OpenCL 2.0" is not a value :).

Reply:
Yes, I thought about defining the encoding also while making this PR, but I didn't want to change too much, since the encoding is also not precisely defined for `__OPENCL_C_VERSION__`.

For this PR, how about changing "any implementation-defined integer value grater than OpenCL 2.0" to "any implementation-defined integer value greater than `CL_VERSION_2_0`"?  That's still not as precise as I would like it to be but I think it's better?

I'll make this change.

Reply:
I changed this to: "any implementation-defined integer value greater than 200, reflecting OpenCL 2.0".  Let me know what you think - thanks!